### ISSUE TITLE
Evaluator thread safety

### ISSUE THREAD

**Issue by justinchuby** [microsoft/onnx-script#297]
Which means no global Evaluator, just to be safe when we do distributed stuff.

**Comment by gramalingam** [comment]
Yes, we need to move some of the globals into thread-local if/when we want to this
